### PR TITLE
[W5.11][T13-2]Cai Xiaolin

### PR DESCRIPTION
**Pull request by xllx1**

### CODE REVIEW

**Comment by HansKoh** [comment]
Excellent enhancements! You have followed Code Style Standard very well, good job!

**Comment by E0191729** [comment]
New Command works well and the code is written well. Good job!

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 1]
```diff
@@ -1,11 +1,6 @@
 package seedu.addressbook.data.person;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
```
Comment (ewaldhew) [inline src/seedu/addressbook/data/person/UniquePersonList.java]: Collapsing imports is against our coding standard. Make sure you set IntelliJ to not do this: Preferences > Editor > Code Style > Java > Imports tab set Class count to use import with '*'

File: test/expected.txt [hunk 2]
```diff
@@ -86,53 +92,62 @@
...
+||
+|| 5 persons listed!
+|| ===================================================
+|| Enter command: || [Command entered:  sort]
+|| 	1. Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street Tags:
+|| 	2. Betsy Choo Tags: [secretive]
+|| 	3. Charlie Dickson Email: charlie.d@nus.edu.sg Address: 333, gamma street Tags: [school][friends]
+|| 	4. Dickson Ee Phone: 444444 Address: 444, delta street Tags: [friends]
+|| 	5. Esther Potato Phone: 555555 Email: esther@not.a.real.potato Tags: [tubers][starchy]
+||
```
Comment (ewaldhew) [inline test/expected.txt]: Good job for updating tests but the test here doesn't tell us anything - the output is the same as `list` so we don't really know if `sort` is working or not. A better test would start off with a list that's not sorted.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+
+import java.util.Collections;
+import java.util.Comparator;
```
Comment (ewaldhew) [inline src/seedu/addressbook/commands/SortCommand.java]: Was this import used here 😕? Avoid adding extraneous imports

File: test/expected.txt [hunk 4]
```diff
@@ -46,26 +48,30 @@
 || 
 || 0 persons listed!
 || ===================================================
+|| Enter command: || [Command entered:  sort]
+||
+|| 0 persons listed!
+|| ===================================================
 || Enter command: || [Command entered:  add wrong args wrong args]
-|| Invalid command format! 
+|| Invalid command format!
```
Comment (ewaldhew) [inline test/expected.txt]: Whitespace changes like this cause your tests to fail and also add unrelated changes to the PR. Always best to double check before committing.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 5]
```diff
@@ -81,6 +76,11 @@ public UniquePersonList(UniquePersonList source) {
         return Collections.unmodifiableList(internalList);
     }
 
+    public List<ReadOnlyPerson> sortedListView() {
```
Comment (ewaldhew) [inline src/seedu/addressbook/data/person/UniquePersonList.java]: Missing header comment. The name of this method a little ambiguous - so does it actually sort the list or just return a sorted view of the list, without touching the original data? Seems like the former here.

**Review by ewaldhew — COMMENTED** [review]
Good effort but take note of various improvements. Close PR after reading
